### PR TITLE
Feature/ltbi results processing

### PR DESCRIPTION
**Pull request by yaqi7**
Hi following is the script of calculating reduction for U5 hh and PLWHIV:
https://github.com/ihmeuw/vivarium_csu_ltbi/blob/feature/ltbi_results_processing/src/vivarium_csu_ltbi/tools/ltbi_result_process.py

### CODE REVIEW

File: src/vivarium_csu_ltbi/tools/ltbi_result_process.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    result['3HP_averted'].fillna(0, inplace = True) # in case 0 denominator 
+    result['6H_averted'] = (result.loc[result['ltbi_treatment_scale_up.scenario'] == 'baseline', 'value'] - result.loc[result['ltbi_treatment_scale_up.scenario'] == '6H_scale_up', 'value'])*100/result.loc[result['ltbi_treatment_scale_up.scenario'] == 'baseline', 'value']
+    result['6H_averted'].fillna(0, inplace = True) # in case 0 denominator
+    final_df = result.reset_index()[['input_draw_number','3HP_averted','6H_averted']].drop_duplicates()
+    addElement(data, country, measure, people, '6H_averted', calculate_mean_ci(final_df, '6H_averted')[0], calculate_mean_ci(final_df, '6H_averted')[1], calculate_mean_ci(final_df, '6H_averted')[2])
+    addElement(data, country, measure, people, '3HP_averted', calculate_mean_ci(final_df, '3HP_averted')[0], calculate_mean_ci(final_df, '3HP_averted')[1], calculate_mean_ci(final_df, '3HP_averted')[2])
+    return
+
+def process_data(df_total, country, measure, pattern):
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/ltbi_result_process.py:71]: I believe this function work for both age_10 & age_100, as well as other averted measures like DALYs and deaths. You can change `df_100_incidence` to `df_outcome` as we might report merged ages %averted other than incidence.

File: src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    new_data = df_data.stack().reset_index().rename(columns = {0:'value', 'level_2' : 'information'})
+    new_data['age_group'] = new_data["information"].str.extract('^.*age_group_([a-z]*_?[a-z]*[0-9]*_?t?o?_?[0-9]*)', expand=True)
+    new_data['hh_status'] = new_data["information"].str.extract("^.*_([a-z]*)_to_hhtb")
+    #calculate proportion
+    num_u5 = new_data[new_data['age_group'].isin(['early_neonatal_', 'late_neonatal_', 'post_neonatal_', '1_to_4'])]["value"].sum()
+    df_u5= new_data[new_data['age_group'].isin(['early_neonatal_', 'late_neonatal_', 'post_neonatal_', '1_to_4'])].groupby("hh_status").sum().reset_index()
+    num_u5_hhtb = df_u5.loc[df_u5["hh_status"] == "exposed","value"][0]
+    hhtb_prop = num_u5_hhtb/num_u5
+    #rescale to national level
+    pop = get_population(location_id= location_dict.get(country),age_group_id=1,sex_id=3,gbd_round_id=5,
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py:34]: It seems like you were using GBD 2017 both sexes, under-5 national population, i wonder is that match to the proportion you calculated? To be more specific, which scenario, sex, and year are you picking from sim person-time measure to calculate the proportion? For rescaling of `u5_hhtb` to national level population, i used 2019 sim baseline results with sex stratification.

Comment (yaqi7) [reply src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py:34]: I used all year info and counts measure.

File: src/vivarium_csu_ltbi/tools/ltbi_result_process.py (hunk 1 above)
Comment (yaqi7) [reply src/vivarium_csu_ltbi/tools/ltbi_result_process.py:71]: Good point, name updated in following link
https://github.com/ihmeuw/vivarium_csu_ltbi/blob/feature/ltbi_results_processing/src/vivarium_csu_ltbi/tools/ltbi_result_process.py

**Review by yongqx — COMMENTED** [review]
@yaqi7 You probably need to change the `mergeCountry` to a load data function since the default output csv under J will be merged ages results, which means we don't have to manually add them up.

**Comment by yaqi7** [comment]
> @yaqi7 You probably need to change the `mergeCountry` to a load data function since the default output csv under J will be merged ages results, which means we don't have to manually add them up.

@yongqx Thank you for reminding. I'll revise my code to upload data once given output data filepath.

File: src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	year_id = 2017).population[0]
+    num_u5_pop = pop*hhtb_prop
+    data.get("country").append(country)
+    data.get("num_u5_pop").append(num_u5_pop)
+    data.get("hhtb_prop(%)").append(hhtb_prop*100)
+
+if __name__ == '__main__':
+    result_dir = '/ihme/costeffectiveness/results/vivarium_csu_ltbi/'
+    loc_100_filePath = {
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py:43]: Please update this to latest file path before we merge it!
`/share/costeffectiveness/results/vivarium_csu_ltbi/no_3hp_babies_{age}/{country}/{latest_date}/output.hdf`

File: src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        'ethiopia_end_100': result_dir + 'updated-input-data-end-100/ethiopia/2020_01_03_14_22_51/output.hdf',
+        'india_end_100': result_dir + 'updated-input-data-end-100/india/2020_01_03_14_23_42/output.hdf',
+        'peru_end_100': result_dir + 'updated-input-data-end-100/peru/2020_01_03_14_23_33/output.hdf',
+        'philippines_end_100': result_dir + 'updated-input-data-end-100/philippines/2020_01_03_14_23_28/output.hdf',
+        'southAfrica_end_100': result_dir + 'updated-input-data-end-100/south_africa/2020_01_03_14_23_36/output.hdf'
+    }
+    loc_10_filePath = {
```
Comment (yongqx) [inline src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py:50]: Same path format for age_end_10.

File: src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py (hunk 3 above)
Comment (yaqi7) [reply src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py:43]: Okay, updated

File: src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py (hunk 4 above)
Comment (yaqi7) [reply src/vivarium_csu_ltbi/tools/number_and_percent_under5_with_hh_AcTB.py:50]: updated, thanks

**Pull request merged**